### ISSUE TITLE
Array inputs to contract functions lead to Bignum errors

### ISSUE THREAD

**Issue by christianlundkvist** [ConsenSys/truffle#13]
I'm testing this contract
```
contract ArrayTest {
    
    uint public x;

    function testArray(uint[] inputArray) {
    x = inputArray[0];
    }
}
```
with test
```
contract 'ArrayTest', (accounts) ->

  it "should assert true", (done) -> 
    arr = ArrayTest.at(ArrayTest.deployed_address)
    arr.testArray([1,2,3,4]).then () ->
      assert.isTrue(true)
      done()
```
It gives the error message below:
```
Unhandled rejection BigNumber Error: new BigNumber() not a number: [object Object]
  at raise (/usr/lib/node_modules/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:1177:25)
  at /usr/lib/node_modules/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:1165:33
  at new BigNumber (/usr/lib/node_modules/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:193:67)
  at new BigNumber (/usr/lib/node_modules/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:203:69)
  at toBigNumber (/usr/lib/node_modules/truffle/node_modules/web3/lib/utils/utils.js:328:12)
  at Object.toTwosComplement (/usr/lib/node_modules/truffle/node_modules/web3/lib/utils/utils.js:339:21)
  at [object Object].formatInputInt [as _inputFormatter] (/usr/lib/node_modules/truffle/node_modules/web3/lib/solidity/formatters.js:41:38)
  at [object Object].SolidityType.formatInput (/usr/lib/node_modules/truffle/node_modules/web3/lib/solidity/coder.js:83:17)
  at [object Object].SolidityCoder._formatInput (/usr/lib/node_modules/truffle/node_modules/web3/lib/solidity/coder.js:161:36)
  at /usr/lib/node_modules/truffle/node_modules/web3/lib/solidity/coder.js:187:21
  at Array.map (native)
  at [object Object].SolidityCoder.encodeParams (/usr/lib/node_modules/truffle/node_modules/web3/lib/solidity/coder.js:186:32)
  at [object Object].SolidityFunction.toPayload (/usr/lib/node_modules/truffle/node_modules/web3/lib/web3/function.js:69:52)
  at [object Object].SolidityFunction.sendTransaction (/usr/lib/node_modules/truffle/node_modules/web3/lib/web3/function.js:131:24)
  at [object Object].SolidityFunction.execute (/usr/lib/node_modules/truffle/node_modules/web3/lib/web3/function.js:208:37)
  at [object Object].<anonymous> (/usr/lib/node_modules/truffle/node_modules/ether-pudding/build/ether-pudding.js:132:23)
  at [object Object].<anonymous> (/usr/lib/node_modules/truffle/node_modules/ether-pudding/build/ether-pudding.js:249:21)
  at [object Object].tryCatcher (/usr/lib/node_modules/truffle/node_modules/bluebird/js/main/util.js:24:31)
  at [object Object].ret [as testArray] (<anonymous>:14:23)
  at Context.<anonymous> (/home/chris/dev/balanc3/test/array_test.coffee:6:14)
  at Test.Runnable.run (/usr/lib/node_modules/truffle/node_modules/mocha/lib/runnable.js:233:15)
  at Runner.runTest (/usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:390:10)
  at /usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:473:12
  at next (/usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:315:14)
  at /usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:325:7
  at next (/usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:260:23)
  at Immediate._onImmediate (/usr/lib/node_modules/truffle/node_modules/mocha/lib/runner.js:292:5)
  at processImmediate [as _immediateCallback] (timers.js:367:17)
```
When I try the same contract & function in Node with `web3` it works.

**Comment by christianlundkvist** [comment]
I tested both with testRPC and geth in the backend. My Node code is here (using testrpc):
```
var web3 = require('web3')

web3.setProvider(new web3.providers.HttpProvider('http://localhost:8545'))

var abi = [{"constant":true,"inputs":[],"name":"x","outputs":[{"name":"","type":"uint256"}],"type":"function"},{"constant":false,"inputs":[{"name":"inputArray","type":"uint256[]"}],"name":"testArray","outputs":[],"type":"function"}]

var code = "606060405260d88060116000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900480630c55699c146041578063d0f126df14606057603f565b005b604a60045060cf565b6040518082815260200191505060405180910390f35b60ab60048035906020019060040180359060200191919080806020026020016040519081016040528093929190818152602001838360200280828437820191505050505050905060ad565b005b8060008151811015600257906020019060200201516000600050819055505b50565b6000600050548156"

//var acct0 = '0x7792e497c7f7de41bb63cdf25f7569440aeeb642'
var acct0 = undefined

hash = web3.eth.sendTransaction({from: acct0, data: code})
addr = web3.eth.getTransactionReceipt(hash).contractAddress
arrContract = web3.eth.contract(abi).at(addr)
arrContract.testArray([1,2,3,4]) // sets x to inputArray[0]
console.log(arrContract.x().toNumber()) // prints "1"
```

**Comment by tymat** [comment]
Your testArray function doesn't return anything.  

You will need to call the getter function for x 

```
 arr.x.call.then (s) ->
```

**Comment by christianlundkvist** [comment]
@tymat I tried calling the getter function and logging the value but the crash happens at `arr.testArray([1,2,3,4])`. What's the versioning issue? Is there a workaround?

**Comment by tcoulter** [comment]
This is fixed in the latest Pudding (0.0.9). To update Truffle, do `npm uninstall -g truffle` and then reinstall it via `npm install -g truffle`.
